lib/ownership_determination.js

```diff
@@ -1,10 +1,8 @@
-const NyplSourceMapper = require('discovery-store-models/lib/nypl-source-mapper')
+// NYPL item ids start with an 'i'
+const NYPL_ITEM_ID_PATTERN = /^i\d+/
```

Comment:
Why this hardcoded change?

Reply:
It's a pragmatic choice. The process of deriving owner from an identifier has been made async. It was previously sync because it was using a request-sync library, which was always dangerous. Now that it's async, anything that needs to derive owner from an identifier using the official method is necessarily async. This specific use of the mapper util (`isItemNyplOwned`) is used throughout the codebase in places that would be awkward to change to async. I think this specific use of the nypl-source-mapper util is unnecessary because the nypl-source-mapper util is primarily meant to future proof as we add/remove partner institutions; A method like `isItemNyplOwned` depends on the identifier convention for NYPL, which is considerably more stable. Were we to change the identifier convention of NYPL, this would be one of several places we'd need to update, but I think 1) that's unlikely to happen in any known time and 2) implementing it as a sync regex check allows us to change far less in this PR, which is mainly about dependency updates.